.github/workflows/integration.yml

```diff
@@ -20,13 +20,23 @@ jobs:
       - uses: actions/setup-go@v4
         with:
           go-version: ${{ env.GO_VERSION }}
+      - run: sudo chown -R $(id -u):$(id -g) /root
+      - uses: actions/cache@v3
+        with:
+          path: |
+            ~/go/pkg/mod
+            ~/.cache/go-build
+            /root/go/pkg/mod
+            /root/.cache/go-build
+          key: go-integration-${{ hashFiles('scripts/versions') }}-${{ hashFiles('scripts/github-actions-setup') }}
```

Comment:
and, if you want hashes of these files to be used as cache key input, there's a `cache-dependency-path` parameter described in actions/setup-go README.md (if unset, it looks for `go.sum` file in `$GITHUB_WORKSPACE`)

Something like this should work:

```yml
    cache-dependency-path: |
        ./go.sum
        ./scripts-versions
        ./scripts-github-actions-setup
```

OTOH, I think that changing a version of a tool we download should not invalidate go cache, so this looks like not being necessary.